buffer/src/test/java/io/netty/buffer/PoolArenaTest.java

```diff
@@ -107,4 +110,22 @@ public final void testAllocationCounter() {
         Assert.assertEquals(1, metric.numNormalDeallocations());
         Assert.assertEquals(1, metric.numNormalAllocations());
     }
+
+    @Test
+    public void testDirectArenaMemoryCopy() {
+        PooledByteBuf<ByteBuffer> src = unwrapIfNeeded(PooledByteBufAllocator.DEFAULT.directBuffer(512));
+        PooledByteBuf<ByteBuffer> dst = unwrapIfNeeded(PooledByteBufAllocator.DEFAULT.directBuffer(512));
```

Comment:
@njhill this may produce leaks as you loose the "wrapper".

You need to do something like this:

```java
ByteBuf src =PooledByteBufAllocator.DEFAULT.directBuffer(512);
ByteBuf dst = PooledByteBufAllocator.DEFAULT.directBuffer(512);

// This causes the internal reused ByteBuffer duplicate limit to be set to 128
dst.writeBytes(ByteBuffer.allocate(128));
         
// Ensure internal ByteBuffer duplicate limit is properly reset (used in memoryCopy non-Unsafe case)
PooledByteBuf<ByteBuffer> pooledSrc = unwrapIfNeeded(src);
PooledByteBuf<ByteBuffer> pooledDst = unwrapIfNeeded(dst);

pooledDst.chunk.arena.memoryCopy(pooledSrc.memory, 0, pooledDst, 512);

src.release();
dst.release();
```

Reply:
@normanmaurer sorry I realized this after pushing but was already falling asleep :) now fixed